operator/v1/types_ingress.go

```diff
@@ -585,6 +591,52 @@ type EndpointPublishingStrategy struct {
 	NodePort *NodePortStrategy `json:"nodePort,omitempty"`
 }
 
+// ClientCertificatePolicy describes the policy for client certificates.
+// +kubebuilder:validation:Enum=Required;Optional
+type ClientCertificatePolicy string
+
+const (
+	// ClientCertificatePolicyRequired indicates that a client certificate
+	// should be required.
+	ClientCertificatePolicyRequired ClientCertificatePolicy = "Required"
+
+	// ClientCertificatePolicyOptional indicates that a client certificate
+	// should be requested but not required.
+	ClientCertificatePolicyOptional ClientCertificatePolicy = "Optional"
+)
+
+// ClientTLS specifies TLS configuration to enable client-to-server
+// authentication, which can be used for mutual TLS.
+type ClientTLS struct {
+	// clientCertificatePolicy specifies whether the ingress controller
+	// requires clients to provide certificates.  This field accepts the
+	// values "Required" or "Optional".
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ClientCertificatePolicy ClientCertificatePolicy `json:"clientCertificatePolicy"`
+
+	// clientCA is a reference to a configmap containing the PEM-encoded CA
+	// certificate bundle that should be used to verify a client's
+	// certificate.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ClientCA configv1.ConfigMapNameReference `json:"clientCA"`
```

Comment:
I had in mind that the configmap would be in the `openshift-ingress` namespace, but that's a bad habit we need to get away from; the configmap should be in `openshift-config`, and the consumer (ingress operator) should copy the configmap to its namespace (`openshift-ingress`).

Reply:
Updated.